### PR TITLE
Add G533 support

### PR DESCRIPTION
**Pull request by rkapl**
Hi,
I have modified the library to support G533 (cheaper version of G933) and fixed some unrelated issues. It at least reports the charging status, and I've randomly tested the other features too.
Thanks, Roman Kapl

### CODE REVIEW

**Comment by ashkitten** [comment]
Hi! Thank you for your interest in the project! I have no time to look at it today but I'll take a look very soon. From a quick overview it looks good, so should not be an issue!

**Comment by ashkitten** [comment]
Looking at it seems fine. @rkapl can you please run `rustfmt-nightly` on the code before I merge?

**Comment by ashkitten** [comment]
The `list` subcommand will now try to get a 4-part name, when there are only 3 parts, resulting in this:
```
Device 3-1: Logitech G933 Gaming Wireless Headsetadset
```
Please use the function in my notes to get the length of the name.

File: libg933/src/lib.rs [hunk 1]
```diff
@@ -255,7 +255,8 @@ impl Device {
         let length = self.raw_request(&request)?[4];
 
         let mut name = String::new();
-        for i in 0..length / 10 {
+        let name_parts = ((length - 1) / 10) + 1;
```
Comment (ashkitten) [inline libg933/src/lib.rs]: What is this? It gets the wrong number of parts for the G933, returning 4 instead of 3

Comment (rkapl) [reply libg933/src/lib.rs]: I had my name chopped of, because length/10 was rounding it down. What is the device name of G933 headset?

Comment (ashkitten) [reply libg933/src/lib.rs]: The device name is 25 characters long, it's in the notes

**Comment by rkapl** [comment]
I've ammended all the commits by running rustfmt.

File: libg933/src/lib.rs (hunk 1 above)
Comment (ashkitten) [reply libg933/src/lib.rs]: oops, it's actually 37 characters long, because hex. but it's also 16 characters returned in the message...

whoops...

Comment (ashkitten) [reply libg933/src/lib.rs]: Just change the 10 to a 16 and it should be fine :+1:

**Comment by ashkitten** [comment]
Your rustfmt might be out of date, or it didn't get applied at all. I'm still seeing many formatting inconsistencies

File: libg933/src/lib.rs [hunk 2]
```diff
@@ -265,12 +265,13 @@ impl Device {
...
     /// Get device name
     pub fn get_device_name(&mut self) -> Result<String, Error> {
-        let request = [0x11, 0xff, 0x03, 0x01];
-        let length = self.raw_request(&request)?[4];
+        let length = self.feature_request(FEATURE_DEVNAME, &[0x01])?[4];
 
         let mut name = String::new();
-        let name_parts = ((length - 1) / 10) + 1;
-        for i in 0..name_parts {
+        let part_size = 16;
```
Comment (ashkitten) [inline libg933/src/lib.rs]: change to a const or remove :)

**Comment by rkapl** [comment]
Yes, I have old rustfmt, sorry for that (did not think it will matter). However, I won't be able to fix that up until thursday, when I can get proper internet and update the whole rust.

**Comment by ashkitten** [comment]
gotcha - it's fine by me to wait

**Comment by ashkitten** [comment]
@rkapl are you prepared to format and merge this?

**Comment by rkapl** [comment]
Yes, I am just working on it right now. Is ``rustfmt-nightly v0.4.0`` the right thing? 
Btw, I'will also add the discharge/charge maps for G533 from Logitech Gaming.

**Comment by ashkitten** [comment]
Yes, that is the correct version. You can also use rustfmt-preview, the rustup component.

**Comment by rkapl** [comment]
I've (hopefully), fixed the formatting, but I am having problem obtaining proper discharge curves for my headphones. I suppose the CVS files are converted from the Logitech PowerProfiles? What's the relation? Thanks.

**Comment by ashkitten** [comment]
iirc i used vim to do some math with the discharge curves to put them on a proper scale. can't remember what it was, but if you want to make a script that does it be my guest.

**Comment by ashkitten** [comment]
It's a map of voltages to battery percentages. If I'm remembering correctly, the original xml file had headers that mentioned what each column corresponded to, you'll want to format the voltages to integers and the percentages to 2 decimal places to match what I got. You might need to do some division to get the right type of value.

**Comment by rkapl** [comment]
Well, the discharge curve maps V->discharge mAh. I thought I just could divide discharge values by maximum discharge and get percanteges, but I am puzzled by values in the map that stick out. For example look at V=3.886:

time (ignored)| V | mAh
-- | -- | --
45 | 3.967 | 50.31222222
45.5 | 3.967 | 50.87138889
46 | 3.967 | 51.43472222
46.5 | 3.886 | 51.97861111
47 | 3.904 | 52.49111111

Or I could go by time, so I am asking, since you've already done the conversion once.

**Comment by ashkitten** [comment]
If you need extra help you'll need to paste the discharge curve xml somewhere, since my hard drive failed and I lost my Windows partition, so I don't have the LGS files anymore.

**Comment by rkapl** [comment]
Here you go: https://rkapl.cz/uploads/logitech/ (A66 - my headset, A5B - yours)

**Comment by ashkitten** [comment]
Okay so it took an hour or so of trying to figure this out, BUT

basically, just copy the discharging csv into a new file in vim, and execute this command:

```
:%s#.*,\(.\{-}\),\(.*\)#\=printf("%04d,%.2f", float2nr(str2float(submatch(1)) * 1000), ((str2nr(submatch(2)) - 4070) * 100) / -4070.0
```

but change 4070 to the maximum capacity of your battery in mWh from the specs sheet (http://support.logitech.com/en_us/product/g933-7-1-surround-sound-gaming-headset/specs for mine)

**Comment by ashkitten** [comment]
as for the charging curves, those are already mapped to SoC values so you just want to format those to the correct number of decimals

**Comment by rkapl** [comment]
Yeah, I believe that's what I've done (but in libreoffice calc). But the problem is that the discharge values jump around (or voltage values jump if you are looking from a different angle). See the extract from the table i've sent. Maybe its not a problem?

**Comment by ashkitten** [comment]
That's not what I get.

Try using

```
:%s#.*,\(.\{-}\),\(.*\)#\=printf("%04d,%.2f", float2nr(str2float(submatch(1)) * 1000), ((str2nr(submatch(2)) - 1100) * 100) / -1100.0)
```

in vim

**Comment by rkapl** [comment]
No, that's exactly what you get too :) Sorry to pester you about it, I just want to check that we are doing it right. 

What I am worried are the discontinuities you see in the graphs (you have one in notes, I've added both to https://rkapl.cz/uploads/logitech/ ). For me, that caused the charge level to jump from 90% to 80% and back in mere seconds (see the graph). So I suppose you did not any filtering on it?

Anyway, I've commited the curve as it is, I might add some more clever filtering in the future, to get rid of these problems.

**Comment by ashkitten** [comment]
That is interesting that that is what is happening. Maybe we can do some filtering, but for now as long as it works it's good with me.

**Comment by rkapl** [comment]
Could you re-do the merge? You've managed to merge the version without rustfmt (i've ammended that). Sorry for the trouble.

**Comment by ashkitten** [comment]
I'll just re-run rustfmt, no big deal.

**Pull request merged**